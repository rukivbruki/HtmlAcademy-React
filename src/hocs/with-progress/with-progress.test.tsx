import * as React from "react";
import * as renderer from "react-test-renderer";
import withProgress from './with-progress';

interface Props {
  forwardedRef?: React.RefObject<HTMLVideoElement>;
}

const MockComponent = ({forwardedRef}: Props) => <video ref={forwardedRef}/>;

const MockComponentWithRef = React.forwardRef((props, ref: React.RefObject<HTMLVideoElement>) => {
  return <MockComponent {...props} forwardedRef={ref} />;
});

MockComponentWithRef.displayName = `test`;

const MockComponentWrapped = withProgress(MockComponentWithRef);

it(`withProgress renders correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped setActivePlayer={() => void 0}/>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
