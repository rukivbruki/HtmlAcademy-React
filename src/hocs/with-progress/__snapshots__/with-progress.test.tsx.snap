// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withProgress renders correctly 1`] = `<video />`;
